src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -21,11 +21,35 @@ namespace ray {
 namespace gcs {
 
 GcsResourceManager::GcsResourceManager(
+    instrumented_io_context &io_context,
     std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
     ClusterResourceManager &cluster_resource_manager)
-    : gcs_table_storage_(gcs_table_storage),
+    : io_context_(io_context),
+      gcs_table_storage_(gcs_table_storage),
       cluster_resource_manager_(cluster_resource_manager) {}
 
+void GcsResourceManager::Update(std::shared_ptr<const syncer::RaySyncMessage> message) {
+  // Make sure thread safety.
+  io_context_.post(
+      [this, message]() {
+        rpc::ResourcesData resources;
+        resources.ParseFromString(message->sync_message());
+        resources.set_node_id(message->node_id());
+        auto node_id = NodeID::FromBinary(message->node_id());
+        if (message->component_id() == syncer::RayComponentId::RESOURCE_MANAGER) {
```

Comment:
A bit advice about naming: maybe use `UsageID` or `Usecase` or something like this intead of `RayComponentId`? Cause `ray component` brings me up `GCS/Raylet/...`, and the sync message is used for some usage/use case, not a concrete component.

Reply:
Hmmm, I feel component id is better. At least you'll know it's a ray component and it's doing the syncing for the components.

UsageID/Usecase  is not that clear about what's the thing here.

Reply:
Maybe SyncMessageType?